OverlayPlugin.Core/MemoryProcessors/FFXIVMemory.cs

```diff
@@ -10,10 +10,11 @@ public class FFXIVMemory
     {
         public event EventHandler OnProcessChange;
 
-        private ILogger logger;
-        private Process process;
-        private IntPtr processHandle;
-        private FFXIVRepository repository;
+        private readonly ILogger logger;
+        private volatile Process process;
+        private volatile IntPtr processHandle;
+        private readonly FFXIVRepository repository;
+        private readonly Object _mutex = new Object();
```

Comment:
This vairable name should be more clear, like `processMutex` or `processLock` which I prefer more.